### ISSUE TITLE
OpenCL Bus error

### ISSUE THREAD

**Issue by ai7603** [microsoft/LightGBM#2550]
I write a test python script using GPU
but there is something wrong
the python script:
`
import lightgbm as lgb
import numpy as np

params = {'max_bin': 63,
'num_leaves': 255,
'learning_rate': 0.1,
'tree_learner': 'serial',
'task': 'train',
'is_training_metric': 'false',
'min_data_in_leaf': 1,
'min_sum_hessian_in_leaf': 100,
'ndcg_eval_at': [1,3,5,10],
'sparse_threshold': 1.0,
'device': 'gpu',
}

train_x = np.array([[1],[1],[0],[1],[1],[0],[0],[0]])
train_y = np.array([1,1,0,1,1,0,0,0])
val_x = np.array([[1],[0],[1],[1]])
val_y = np.array([1,0,1,1])

test_x = [1,0,1,0]

tr_dt = lgb.Dataset(train_x, label=train_y)
val_dt = lgb.Dataset(val_x, label=val_y)

gbm = lgb.train(params, tr_dt, 4, valid_sets=[tr_dt,val_dt])
`
but when I run it, BUS ERROR:
`
[LightGBM] [Info] This is the GPU trainer!!
[LightGBM] [Info] Total Bins 2
[LightGBM] [Info] Number of data: 8, number of used features: 1
[LightGBM] [Info] Using GPU Device: GeForce GTX 1080, Vendor: NVIDIA Corporation
[LightGBM] [Info] Compiling OpenCL Kernel with 16 bins...
Bus error
`
what's happen and what can I do to deal with it?
Thanks

**Comment by StrikerRUS** [comment]
@ai7603 Please make sure you're using the latest version of LightGBM and drivers for your GPU.

I remember we had a fix for cases with extremely small number of bins recently.

Ping @huanzhang12 for further investigation.
